zarrs/src/array_subset/iterators/contiguous_indices_iterator.rs

```diff
@@ -86,15 +70,18 @@ impl ContiguousIndices {
         }
         // SAFETY: each element is initialised
         unsafe { shape_out.set_len(array_shape.len()) };
-        // SAFETY: The length of shape_out matches the subset dimensionality
-        let subset_contiguous_start = unsafe {
-            ArraySubset::new_with_start_shape_unchecked(subset.start().to_vec(), shape_out)
-        };
+        let ranges = subset
+            .start()
+            .iter()
+            .zip(shape_out)
+            .map(|(&st, sh)| st..(st + sh))
+            .collect::<Vec<_>>();
+        let subset_contiguous_start = ArraySubset::new_with_ranges(&ranges);
```

Comment:
I used this pattern a number of times to minimize the amount of new errors we need to deal with i.e., we have a start and a shape, either not checked previously in the code or, as here, checked above for having the same length, and then we use `new_with_ranges`.  So the `ranges` creation could become a utility but not sure the best place to put it.

Reply:
See the main review comment